processing/src/main/java/io/druid/segment/DoubleDimensionIndexer.java

```diff
@@ -0,0 +1,256 @@
+  )
+  {
+    final int dimIndex = desc.getIndex();
+    class IndexerLongColumnSelector implements LongColumnSelector
+    {
+      @Override
+      public long get()
+      {
+        final Object[] dims = currEntry.getKey().getDims();
+
+        if (dimIndex >= dims.length) {
+          return 0L;
+        }
+
+        double doubleValue = (Double) dims[dimIndex];
+        return (long) doubleValue;
+      }
+
+      @Override
+      public void inspectRuntimeShape(RuntimeShapeInspector inspector)
+      {
+        // nothing to inspect
+      }
+    }
+
+    return new IndexerLongColumnSelector();
+  }
+
+  @Override
+  public FloatColumnSelector makeFloatColumnSelector(
+      IncrementalIndexStorageAdapter.EntryHolder currEntry, IncrementalIndex.DimensionDesc desc
+  )
+  {
+    final int dimIndex = desc.getIndex();
+    class IndexerFloatColumnSelector implements FloatColumnSelector
+    {
+      @Override
+      public float get()
+      {
+        final Object[] dims = currEntry.getKey().getDims();
+
+        if (dimIndex >= dims.length) {
+          return 0.0f;
+        }
+
+        double doubleValue = (Double) dims[dimIndex];
+        return (float) doubleValue;
+      }
+
+      @Override
+      public void inspectRuntimeShape(RuntimeShapeInspector inspector)
+      {
+        // nothing to inspect
+      }
+    }
+
+    return new IndexerFloatColumnSelector();
+  }
+
+  @Override
+  public ObjectColumnSelector makeObjectColumnSelector(
+      DimensionSpec spec, IncrementalIndexStorageAdapter.EntryHolder currEntry, IncrementalIndex.DimensionDesc desc
+  )
+  {
+    final int dimIndex = desc.getIndex();
+    class IndexerObjectColumnSelector implements ObjectColumnSelector
+    {
+      @Override
+      public Class classOfObject()
+      {
+        return Double.class;
+      }
+
+      @Override
+      public Object get()
+      {
+        final Object[] dims = currEntry.getKey().getDims();
+
+        if (dimIndex >= dims.length) {
+          return DimensionHandlerUtils.ZERO_DOUBLE;
+        }
+
+        return dims[dimIndex];
+      }
+    }
+
+    return new IndexerObjectColumnSelector();
+  }
+
+  @Override
+  public DoubleColumnSelector makeDoubleColumnSelector(
+      IncrementalIndexStorageAdapter.EntryHolder currEntry, IncrementalIndex.DimensionDesc desc
+  )
+  {
+    final int dimIndex = desc.getIndex();
+    class IndexerDoubleColumnSelector implements DoubleColumnSelector
+    {
+      @Override
+      public double get()
+      {
+        final Object[] dims = currEntry.getKey().getDims();
+
+        if (dimIndex >= dims.length) {
+          return 0.0;
+        }
+        return (Double) dims[dimIndex];
+      }
+
+      @Override
+      public void inspectRuntimeShape(RuntimeShapeInspector inspector)
+      {
+        // nothing to inspect
+      }
+    }
+
+    return new IndexerDoubleColumnSelector();
+  }
+
+  @Override
+  public int compareUnsortedEncodedKeyComponents(Double lhs, Double rhs)
```

Comment:
Please mark parameters `@Nullable`, as well as in the implemented interface. As well as in similar methods i. e. checkUnsortedEncodedKeyComponentsEqual, getUnsortedEncodedKeyComponentHashCode.